vue/test/components/select/select.test.js

```diff
@@ -0,0 +1,324 @@
+const assert = require("assert");
+const base = require("../../base");
+
+describe("Select", () => {
+    it("should start with null value", () => {
+        const component = base.getComponent("Select");
+        const componentData = component.vm.$data;
+        assert.strictEqual(componentData.valueData, null);
+    });
+
+    it("should start with no dropdown items", () => {
+        const component = base.getComponent("Select");
+        const dropdownItemsLength = component.findAll(".dropdown-item").length;
+        assert.strictEqual(dropdownItemsLength, 0);
+    });
+
+    it("should display all options labels according to prop", async () => {
+        const component = base.getComponent("Select", {
+            props: {
+                options: [
+                    { value: "option_1", label: "A" },
+                    { value: "option_2", label: "B" },
+                    { value: "option_3", label: "C" }
+                ],
+                value: "option_1"
+            }
+        });
+
+        const dropdownItemsLength = component.findAll(".dropdown-item").length;
+        const dropdownFirstItem = component.findAll(".dropdown-item").at(0);
+        const dropdownSecondItem = component.findAll(".dropdown-item").at(1);
+        const dropdownThirdItem = component.findAll(".dropdown-item").at(2);
+        assert.strictEqual(dropdownItemsLength, 3);
+        assert.strictEqual(dropdownFirstItem.text(), "A");
+        assert.strictEqual(dropdownSecondItem.text(), "B");
+        assert.strictEqual(dropdownThirdItem.text(), "C");
+    });
+
+    it("should update value according to prop", async () => {
+        const component = base.getComponent("Select", {
+            props: {
+                options: [
+                    { value: "option_1", label: "A" },
+                    { value: "option_2", label: "B" },
+                    { value: "option_3", label: "C" }
+                ],
+                value: "option_1"
+            }
+        });
+        const componentData = component.vm.$data;
+        const selectButton = component.get(".select-button");
+        assert.strictEqual(selectButton.text(), "A");
+        assert.strictEqual(componentData.valueData, "option_1");

```

Comment:
We shouldn't do it like this, because it's not equivalent to what we do when using this component. Tests must rely on public API (props, events, visual state, etc) rather on internal data.

Because of this, this must be changed to bind to the `update:value` event and make assertions based on that. On "real world" usage of this component that's what we do when using `sync`, we never look for internal data.